8.0 Adapt sale_rental and sale_order_lot_selection to make them work together

alexis-via opened the pull request:
Adapt sale_rental and sale_order_lot_selection to make them work together from a functional point of view, via the module sale_rental_lot_selection.

In sale_order_lot_selection/model/sale.py, at the end of the file, there are several methods that are inherited and I think it is not necessary (or I don't understand it's usage). So we still need to discuss this:
- either this code is generic and you explain what is it's use
- either this code is specific for a particular project ; in this case, it should be moved elsewhere.

eLBati committed:
[REF] renamed module

eLBati committed:
[REF] start refactoring of sale_line_quantity_properties_based

eLBati committed:
[FIX] descriptions

eLBati committed:
[FIX] PEP8

eLBati committed:
[REF] one model per file

eLBati committed:
[DEL] sale_properties_easy_creation/sale_demo.xml

eLBati committed:
[FIX] refers to old module name

pedrobaeza committed:
[ADD] sale_order_recalculate_prices: Module to allow to recalculate on demand sale order line prices

==================================================
Recalculation of sale order lines prices on demand
==================================================

This module adds a button on sale orders below pricelist that recalculates the
prices of the order lines that contain a product in them.

It is launched manually as a button to get the user decide if he/she wants to
recalculate prices when pricelist is changed.

Usage
=====

Inside a sale order, you can click in any moment a button called
"(Recalculate prices)", that is next to the pricelist selection, to launch
a recalculation of all the prices of the lines, losing previous custom prices.

Known issues / Roadmap
======================

* In a sale order with lot of lines, the recalculation may slow down, because
  sale general data (amount untaxed, amount taxed...) are recalculated for
  each line.

moylop260 committed:
[REF] auto-fix conventions

Commit:
Rename sale_payment_method_automatic_reconcile to  sale_payment_method_automatic_workflow

Because it goes further than only reconciliation

Commit:
Set the module as auto_install

So it installs when both sale_payment_method and sale_automatic_workflow are installed.
This module acts as the glue between them

Commit:
Adapt description and name

Commit:
Add the link between the payment method and the automatic workflow

Commit:
Fix coding errors

Commit:
Fix pep8 in __openerp__

Commit:
Update translations

Commit:
Add translations

guewen committed:
Typo in a help message

guewen committed:
Method returns 2 move lines, add a final 's'

Keep the original method with a deprecation warning to help people eventually
inheriting this method to not lose their minds when migrating their code.

gurneyalex committed:
Add OCA as author of OCA addons

In order to get visibility on https://www.odoo.com/apps the OCA board has
decided to add the OCA as author of all the addons maintained as part of the
association.

guewen committed:
Correct author list, add OCA

guewen committed:
Useless dict returned

guewen committed:
str() not needed

guewen committed:
Add comments on the currency conversion

guewen committed:
Use copy=False on the field rather than copy method

bguillot committed:
[FIX] change called method to confirm sale orders

Commit:
Remove the field sale_order.payment_exists

The computed field was used to hide the button 'View Automatic Payment' when
the sales order has at least one payment. It is now using only a condition on
payment_ids, thanks to @hbrunn for the correct definition: [('payment_ids', '=', [[6, False, []]])]

matthieuchoplin committed:
[FIX] a Many2one in the new api is expecting a recordset

matthieuchoplin committed:
add returns decorator

tafaRU committed:
Add code using by 'git checkout origin/7.0 -- sale_properties_dynamic_fields

tafaRU committed:
Use tag group instead of label tag

tafaRU committed:
Update copyright year

tafaRU committed:
Extract the description in README.rst

tafaRU committed:
Add OCA icon

tafaRU committed:
Remove active key since is deprecated

tafaRU committed:
Add group to view

tafaRU committed:
Extract the description in README.rst

tafaRU committed:
Add OCA icon

tafaRU committed:
Update copyright year

tafaRU committed:
Fix contributors' mails

Commit:
[ADD] Module sale_pricelist_discount

Commit:
[PEP8] Module sale_pricelist_discount

Commit:
[IMP] Refactor and layout improvements

yvaucher committed:
Fix name_search for sourced_by field

When searching by Purchase Order name, domain was ignored

pedrobaeza committed:
Merge pull request #32 from guewen/8.0-sale_automatic_worflow-many-so

Do not propagate sale sale automatic workflow when several orders are grouped in one picking

tafaRU committed:
Fix depends, as m2m property_ids moved from sale_stock to sale_mrp in V8

tafaRU committed:
Remove active key as deprecated

bguillot committed:
[IMP] move file

bguillot committed:
Merge branch '8.0_move_sale_payment_method' into move_sale_payment_method

damdam-s committed:
[MIG] migrate module 'account_invoice_reorder_lines' to v8

esthermm committed:
[ADD] <partner_handbill> new module added

esthermm committed:
[IMP] Function improved

esthermm committed:
[ADD] partner_handbill tests added

esthermm committed:
[ADD] partner_handbill new case for test

oihane committed:
[IMP] <partner_handbill> Added OCA required README.rst

oihane committed:
[IMP] <partner_prospect> Changed *handbill* for *prospect*

oihane committed:
[IMP] <partner_prospect> added sale orders by commercial partner

oihane committed:
[IMP] <partner_prospect> Upgraded tests

oihane committed:
[IMP] <partner_prospect> Fix mistake in README

yvaucher committed:
Add missing default oca icons

yvaucher committed:
Add missing default oca icons

Commit:
[FIX] sale_order_type: Fixed bug when picking is created from a purchase order

damdam-s committed:
[IMP] add relation to sales team

damdam-s committed:
[IMP] add group base.group_multi_salesteams on 'section_id' fields in views

Commit:
[IMP] sale_order_type: Test improved

gurneyalex committed:
Merge pull request #54 from pedrobaeza/8.0-sale_order_recalculate_prices

[ADD] sale_order_price_recalculation: Module to allow to recalculate on demand sale order line prices

eLBati committed:
REF travis for transifex and last updates

tafaRU committed:
Get product aware of the language partner

tafaRU committed:
Name each keyword argument

Commit:
Merge pull request #204 from tafaRU/8.0-sale_order_line_description-fix

[FIX] sale_order_line_description: missing translation for field description_sale

tafaRU committed:
Merge pull request #203 from eLBati/8_add_transifex

REF travis for transifex and last updates

yvaucher committed:
Bump version of sale_order_type

yvaucher committed:
Merge branch 'Daniel-CA-sale_order_type' into 8.0

legalsylvain committed:
[FIX] remove module stock_picking_reorder_lines from unported folder. (the module is now in OCA / stock-logistic-workflow);

tafaRU committed:
Merge pull request #208 from grap/8.0_FIX_unported_folder__stock_picking_reorder_lines

[FIX] remove module stock_picking_reorder_lines from unported folder (the module is now in OCA / stock-logistic-workflow);

oca-transbot committed:
OCA Transbot updated translations from Transifex

oihane committed:
[FIX] <sale_order_type> Invoice state from type not applied in pickings

Commit:
[product_set]copy product set line when
copying product set

Commit:
[product set] add copy in product set name when copying

Commit:
[product_set] upgrde translations

Commit:
remove useless underscore
field labels and helps attributes are translated by default

Commit:
add choice to recompute price only
This commit add the capability to reset sale order line
without renaming sale order lines

Commit:
upgrade translation

oihane committed:
[IMP] <partner_prospect> Required changes in README.rst

oihane committed:
[IMP] <sale_order_type> updated tests

Commit:
Merge pull request #193 from oihane/8.0-partner_handbill

[ADD] partner_prospect

Commit:
update sale_order_price_recalculation doc

Commit:
display recompute button only on quotations

jjscarafia committed:
IMP add exceptions in quotation tree view

Commit:
rename menu 'Product set' to 'Product Set'

Commit:
unit test duplicate product set name
and space triming on tranlsated string

sbidoul committed:
[ADD] addons table in README.md

oihane committed:
[IMP] <partner_prospect> Now takes into account the invoices

alexis-via committed:
Add module sale_rental_lot_selection

Adapt sale_rental and sale_order_lot_selection to make them work together from a functional point of view

alexis-via commented:
You can watch the result of this development in this screencast:
http://people.via.ecp.fr/~alexis/screencast-rental-lot-selection.ogv

eLBati commented:
Thanks @alexis-via 

The `_check_move_state` method is used to check whether, after SO confirmation,  the stock move has been correctly created or not, and to assign it.

See order3 test case: https://github.com/akretion/sale-workflow/blob/8.0-add-sale_rental_lot_selection/sale_order_lot_selection/tests/test_sale_order_lot_selection.py#L210-L213

For example, if user A selects a lot in SO1, then  user B delivers it and user A confirms  SO1, the system warns

> Can't reserve products for lot %s

Commit:
Split functionnalities reset prices/reset desciptions
Move button under sale order lines

pedrobaeza committed:
Merge pull request #214 from anybox/8.0-sale_recalculation-optionnaly-without-name

8.0 sale recalculation optionnaly without name

Commit:
Display reset button on SO when state == sent

pedrobaeza committed:
Merge pull request #219 from anybox/8.0-sale_recalculation-visible_when_sent

Display reset button on SO when state == sent

pedrobaeza committed:
Merge pull request #157 from yvaucher/8.0-fix-pol-name_search-2

Fix name_search for sourced_by field

pedrobaeza committed:
Merge pull request #212 from oihane/8.0-fix-sale_order_type

[FIX] <sale_order_type> Invoice state from type not applied in pickings

Commit:
Merge pull request #216 from adhoc-dev/8.0.add_exceptions_on_qout_tree

IMP add exceptions in quotation tree view

oca-transbot committed:
OCA Transbot updated translations from Transifex

sergio-teruel committed:
[ADD] sale_packaging_price: New module to assign a package price.

pedrobaeza committed:
[IMP] README: Update coveralls badge

sbidoul committed:
[UPD] prefix versions with 8.0

sbidoul committed:
[UPD] prefix versions with 8.0

sbidoul committed:
[UPD] prefix versions with 8.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

eLBati commented:
@alexis-via even after applying  the following diff
http://pastie.org/10476717
to your branch, tests are failing with 

```
2015-10-12 14:04:14,563 23135 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: `   File "/home/elbati/workspace/odoo/demo8/parts/sale-workflow/sale_order_lot_selection/model/sale.py", line 121, in _check_move_state
2015-10-12 14:04:14,563 23135 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: `     line.lot_id.name)
2015-10-12 14:04:14,563 23135 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: ` ValueError: "Can't reserve products for lot 0000010" while evaluating
2015-10-12 14:04:14,563 23135 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: ` u'action_ship_create()'
```

Do you know why?

sbidoul committed:
[MOV] move addons out of __unported__ (they remain not installable)

sbidoul committed:
[MOV] move addons out of __unported__ (they remain not installable)

sbidoul committed:
[MOV] move addons out of __unported__ (they remain not installable)

yajo committed:
Remove duplicated addon.

It can be found in https://github.com/OCA/purchase-workflow/tree/8.0/purchase_order_reorder_lines.

pedrobaeza committed:
Merge pull request #222 from Yajo/remove-purchase-order-reorder-lines

Remove duplicated addon.

StefanRijnhart committed:
[ADD] Default sales incoterm per partner

Commit:
[FIX] Remove substitution markers

alexis-via committed:
Fix auto-configuration of multiple warehouses for rental

Commit:
remove deprecated api.one and ensure there is only one record when copying element

alexis-via committed:
Warehouse setup for rental v2

sergio-teruel committed:
[ADD][8.0] sale_service_project: New module.

sergio-teruel committed:
[ADD][8.0] sale_service_fleet: New module.

sergio-teruel committed:
[ADD][8.0] sale_service_fleet: New module.

sergio-teruel committed:
[ADD][8.0] sale_service_project: Readme change id repo.

Commit:
[RFR] Not a commercial field

rven committed:
[8.0][FIX] Sale order lines not set to draft

When creating a new revision of a cancelled sale order, the sale order is set to the state draft, but the order lines are not updated to draft.
The lines are still in state cancelled and cannot be updated.

pedrobaeza committed:
Merge pull request #213 from anybox/8.0-copy_product_set

8.0 copy product set

sergio-teruel committed:
[FIX] sale_packaging_price: Changes suggested by @yajo and @pedrobaeza.

sergio-teruel committed:
[FIX] sale_packaging_price: Fix flake8.

pedrobaeza committed:
Merge pull request #230 from rven/8.0-sale_line_not_in_draft

[8.0][FIX] Sale order lines not set to draft

tafaRU committed:
Merge pull request #225 from StefanRijnhart/8.0-sale_partner_incoterm

[ADD] Default sales incoterm per partner

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
[UPD] addons table in README.md

alexis-via committed:
Restore initial code in sale_order_lot_selection/model/sale.py

alexis-via commented:
@eLBati I made

alexis-via commented:
@eLBati I restored the part of your code that I didn't understand at the time and that I commented. made a rebase, so you should be able to merge this now.

eLBati commented:
Thanks @alexis-via please also fix tests:

```
2015-10-29 11:49:54,621 16445 INFO demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: test_sale_order_lot_selection (openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection.TestSaleOrderLotSelection)
2015-10-29 11:49:57,046 16445 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: FAIL
2015-10-29 11:49:57,047 16445 INFO demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: ======================================================================
2015-10-29 11:49:57,047 16445 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: FAIL: test_sale_order_lot_selection (openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection.TestSaleOrderLotSelection)
2015-10-29 11:49:57,047 16445 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: Traceback (most recent call last):
2015-10-29 11:49:57,047 16445 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: `   File "/home/elbati/workspace/odoo/demo8/parts/sale-workflow/sale_order_lot_selection/tests/test_sale_order_lot_selection.py", line 208, in test_sale_order_lot_selection
2015-10-29 11:49:57,047 16445 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: `     self.assertEqual(onchange_res['domain']['lot_id'], [('id', 'in', [])])
2015-10-29 11:49:57,047 16445 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: ` AssertionError: Lists differ: [('id', 'in', []), ('product_i... != [('id', 'in', [])]
2015-10-29 11:49:57,047 16445 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: ` 
2015-10-29 11:49:57,047 16445 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: ` First list contains 1 additional elements.
2015-10-29 11:49:57,047 16445 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: ` First extra element 1:
2015-10-29 11:49:57,047 16445 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: ` ('product_id', '=', 20)
2015-10-29 11:49:57,047 16445 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: ` 
2015-10-29 11:49:57,047 16445 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: ` - [('id', 'in', []), ('product_id', '=', 20)]
2015-10-29 11:49:57,047 16445 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: ` + [('id', 'in', [])]
2015-10-29 11:49:57,047 16445 INFO demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: Ran 1 test in 2.427s
2015-10-29 11:49:57,047 16445 ERROR demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection: FAILED
2015-10-29 11:49:57,048 16445 INFO demo_8 openerp.addons.sale_order_lot_selection.tests.test_sale_order_lot_selection:  (failures=1)
2015-10-29 11:49:57,048 16445 ERROR demo_8 openerp.modules.module: Module sale_order_lot_selection: 1 failures, 0 errors
2015-10-29 11:49:57,061 16445 INFO demo_8 openerp.modules.loading: 37 modules loaded in 3.77s, 3086 queries
2015-10-29 11:49:58,017 16445 ERROR demo_8 openerp.modules.loading: At least one test failed when loading the modules.
```

Commit:
[IMP] Remove description from __openerp__.py now that it is in README.rst

pedrobaeza committed:
Merge pull request #155 from credativUK/8.0-sale_pricelist_discount

8.0 sale pricelist discount

carlosdauden committed:
[IMP][8.0] sale_packaging_price: Check list price differences with decimal precision unit price

carlosdauden committed:
[IMP][8.0] sale_packaging_price: Rename sale_stock to sale_order

sergio-teruel committed:
[IMP] sale_service_project: Add project to oca_dependencies.txt

tafaRU committed:
Use split without sep to properly manage multiple spaces, tabs, etc.

carlosdauden committed:
[FIX][8.0] sale_packaging_price: ZeroDivisionError, Translate

carlosdauden committed:
[FIX][8.0] sale_packaging_price: Better message description

sbidoul committed:
[UPD] addons table in README.md

tafaRU committed:
Add compute_formula method to MrpPropertyFormula to use in modules which depending on it

tafaRU committed:
Add translation files

tafaRU committed:
Improve README and update module version according to last OCA conventions

tafaRU committed:
Add category to manifest file

tafaRU committed:
Improve README and manifest file according to last OCA conventions

tafaRU committed:
Add Configuration section

tafaRU committed:
Add help to formula_text field

gurneyalex committed:
Merge pull request #167 from tafaRU/8.0-sale_properties_easy_creation

[8.0] Migration of sale_properties_easy_creation

tafaRU committed:
Clean yml test removing the commented part

tafaRU committed:
Create a copy of context before modifying it as context being a frozen dict

tafaRU committed:
Modify .travis to test sale_properties_dynamic_fields separately

OCA-git-bot committed:
[UPD] addons table in README.md

pedrobaeza committed:
Merge pull request #217 from odoomrp/8.0-fix_prospect

[IMP] <partner_prospect> Now takes into account the invoices

tafaRU committed:
Replace orm.except_orm by exceptions.Warning

tafaRU committed:
Apply some code cleanup

tafaRU committed:
Add pot file

tafaRU committed:
Remove size from vals used to build the dynamic field and avoid backslashes from context as not required

tafaRU committed:
Replace draw by display and update pot file

gurneyalex committed:
Merge pull request #147 from tafaRU/8.0-sale_properties_dynamic_fields-migr

Migration of sale_properties_dynamic_fields

OCA-git-bot committed:
[UPD] addons table in README.md

damdam-s committed:
[FIX] typo

damdam-s committed:
[FIX] according to code review

damdam-s committed:
[FIX] change description

tafaRU committed:
Merge pull request #189 from damdam-s/migrate_account_invoice_reorder_lines

[MIG] migrate module 'account_invoice_reorder_lines' to v8

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA-git-bot committed:
[UPD] addons table in README.md

tafaRU committed:
Merge pull request #172 from akretion/8.0_move_sale_payment_method

8.0 move sale payment method

lepistone committed:
Merge branch 'move-sale-automatic-workflow-exception' into move-sale-automatic-workflow

lepistone committed:
Merge branch 'move-sale-payment-method-automatic-workflow' into move-sale-automatic-workflow

lepistone committed:
Merge branch 'move-sale-payment-method-transaction-id' into move-sale-automatic-workflow

lepistone committed:
Merge branch '8.0' into move-sale-automatic-workflow

tafaRU committed:
Merge pull request #240 from lepistone/move-sale-automatic-workflow

move 4 modules from OCA/e-commerce

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
OCA Transbot updated translations from Transifex

carlosdauden committed:
[FIX][8.0] sale_service_project: Changes suggested by @Yajo

Commit:
sale_automatic_workflow : add default journal on workflow process

carlosdauden committed:
[FIX][8.0] sale_service_project: Create subprojects

carlosdauden committed:
[IMP] sale_service_project: Create subprojects. View tasks from SO

carlosdauden committed:
[FIX] sale_service_project: PEP8

carlosdauden committed:
[FIX][8.0] sale_service_project: Compute task hours. Project parent in task

sergio-teruel committed:
[IMP][8.0] sale_service_project: Remove invoice policy 'analytic'

carlosdauden committed:
[FIX][8.0] sale_service_project: Changes suggested by @pedrobaeza

oca-transbot committed:
OCA Transbot updated translations from Transifex

rafaelbn committed:
Merge pull request #221 from incaser/sale_packaging_price

[ADD] sale_packaging_price: New module to assign a package price.

OCA-git-bot committed:
[UPD] addons table in README.md

tafaRU committed:
Add sale_order_line_variant_description module

carlosdauden committed:
[FIX][8.0] sale_service_project: Button View Tasks don't it show properly

sergio-teruel committed:
[IMP][8.0] sale_service_project: Security ACL

carlosdauden committed:
[FIX][8.0] sale_service_project: Invoice policy depends invoice_on_timesheets option on analytic account

Commit:
sale_automatic_workflow : create invoice for sale with only product service

sergio-teruel committed:
[IMP] sale_service_project: Add works and materials to sale order report

sergio-teruel committed:
[FIX] sale_service_project: Pep8

sergio-teruel committed:
[FIX][8.0] sale_service_project: fix inherit sale report.

sergio-teruel committed:
[FIX][8.0] sale_service_project: fix readme.

oca-transbot committed:
OCA Transbot updated translations from Transifex

eLBati committed:
Merge pull request #250 from tafaRU/8.0-sale_order_line_variant_description-add

Add sale_order_line_variant_description module

OCA-git-bot committed:
[UPD] addons table in README.md

sergio-teruel committed:
[IMP] sale_packaging_price: add material package type

sergio-teruel committed:
[IMP] sale_packaging_price: add material package type model

sergio-teruel committed:
[IMP] sale_packaging_price: Translate

sergio-teruel committed:
[FIX] sale_packaging_price: Changes suggested by @yajo

carlosdauden committed:
[IMP][8.0] sale_packaging_price: Packaging material adds to readme description and demo.

oca-transbot committed:
OCA Transbot updated translations from Transifex

sergio-teruel committed:
[IMP][8.0] sale_service_project: Print tasks and materials in invoice report

sergio-teruel committed:
[IMP] sale_service_project: Adds id attribute in <p> into invoice report

Commit:
sale_payment_method : use partner_invoice instead of partner when add payment

StefanRijnhart committed:
Merge pull request #202 from eLBati/add_sale_order_lot_selection

Add sale order lot selection

OCA-git-bot committed:
[UPD] addons table in README.md

carlosdauden committed:
[IMP][8.0] sale_service_project: Invoice policy simplified. Action Invoice from Task added.

carlosdauden committed:
[FIX] sale_service_project: Flake8. Improved to_invoice selection

carlosdauden committed:
[FIX] sale_service_project: @yajo and @oihane suggestions.

carlosdauden committed:
[FIX] sale_service_project: Better description in wizard.

carlosdauden committed:
[IMP] sale_service_project: Check if task is closed

carlosdauden committed:
[IMP][8.0] sale_service_project: Changes suggested by @yajo

carlosdauden committed:
[IMP][8.0] sale_service_project: Copy works and materials when duplicate SO

carlosdauden committed:
[IMP][8.0] sale_service_project: Task closed control in Sale Order Line

carlosdauden committed:
[IMP][8.0] sale_service_project: Invoice report from tasks. Minor changes.

carlosdauden committed:
[IMP][8.0] sale_service_project: Minor fix

carlosdauden committed:
[FIX] sale_service_project: Rename material_id to product_id

carlosdauden committed:
[IMP][8.0] sale_service_project: Clean up comments and improved readme

eLBati committed:
FIX sale_properties_dynamic_fields tests

As it seems something does wrong with writing mrp.property.group field with '!record' YAML statement

This avoids 'Field name must be 'x_length', False found'

rafaelbn committed:
Merge pull request #226 from incaser/sale_service_project

[ADD][8.0] sale_service_project: New module.

eLBati committed:
FIX You cannot transfer a picking in state 'False'

OCA-git-bot committed:
[UPD] addons table in README.md

rafaelbn committed:
Merge pull request #260 from incaser/8.0-PR-sale_packaging_price

[IMP][8.0]sale_packaging_price: Add material package type

eLBati committed:
FIX update field_id only when changing draw_dynamically field

eLBati committed:
FIX mixed API do not propagate context

carlosdauden committed:
Merge remote-tracking branch 'origin/sale_service_fleet' into sale_service_fleet

Conflicts:
	sale_service_fleet/README.rst
	sale_service_fleet/report/project_report.py

rafaelbn committed:
Merge pull request #266 from eLBati/fix_sale_properties_dynamic_fields_tests

FIX sale_properties_dynamic_fields and sale_order_lot_selection tests

carlosdauden committed:
[IMP][8.0] sale_service_fleet: Improve readme.

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
Merge pull request #254 from acsone/8.0-sale_automatic_workflow_create_invoice-lga

sale_automatic_workflow : create invoice for sale with only service

sbidoul committed:
Merge pull request #262 from acsone/8.0-sale_payment_method_partner_invoice-lga

[8.0] sale_payment_method : use partner_invoice instead of partner

carlosdauden committed:
[IMP][8.0] sale_service_fleet: Changes suggested by @pedrobaeza

carlosdauden committed:
[IMP][8.0] sale_service_fleet: Change string projects to contracts

pedrobaeza committed:
Merge pull request #227 from incaser/sale_service_fleet

[ADD][8.0] sale_service_fleet: New module.

OCA-git-bot committed:
[UPD] addons table in README.md

pedrobaeza committed:
Merge pull request #247 from acsone/8.0-set_journal-lga

[8.0] sale_automatic_workflow : add default journal on workflow process

alexis-via committed:
Add OCA as author

Warning -> UserError
Fix changes in sale_order_lot_section

alexis-via committed:
Adapt tests of sale_order_lot_selection

alexis-via committed:
sale_rental: Add store=True on some fields, in particular 'state'

Add search view for rental
FIX compatibility between sale_rental and sale_order_lot_selection

alexis-via committed:
Add rental state "cancel"

Better string

sebastienbeau committed:
[FIX] fix perf issue with sale_order_lot_selection

alexis-via committed:
Make it work with older version of Odoo 8

jabibi committed:
[FIX] populate the 'Return Stock Move' field; module version

When creating the rental record, 'Return Stock Move' field was not
taking the corresponding reference to the incoming stock.move

Also fixed the version number and added Odoo Community Association (OCA)
to author list

jabibi committed:
[IMP] Display selection label instead of key, for rental display name

jabibi committed:
[FIX] Complete @api.constrains dependencies for rented products

jabibi committed:
[I18N] spanish translation

Commit:
Merge pull request #4 from jabibi/8.0-add-sale_rental_lot_selection

bug fixes, minor improvements and spanish translation

eLBati commented:
@alexis-via something went wrong with this: github says 250+ commits and 659 files changed
